colossus/src/main/scala/colossus/streaming/PipeMultiplexer.scala

```diff
@@ -0,0 +1,130 @@
+package colossus
+package streaming
+
+import controller._
+
+trait MultiStream[K, T] extends Stream[T] {
+  def streamId(t: T): K
+}
+
+case class SubSource[K,T](id: K, stream: Source[T])
+case class SubSink[K,T](id: K, stream: Sink[T])
+
+object Multiplexing {
+
+  def multiplex[K,T](base: Sink[T])(implicit ms: MultiStream[K,T]): Sink[SubSource[K,T]] = {
+    val sources = new BufferedPipe[SubSource[K,T]](10)
```

Comment:
Ah I made this configurable on the demultiplexing side but forgot to do it for multiplexing.  I think that means I also forgot to test backpressure on the sources pipe so I'll fix that as well.